File: Queue/Queue.java
import java.util.*;

class Node{
    int data;
    Node next;

    public Node(int data){
        this.data = data;
        this.next = null;
    }

    public int getData() {return data;}

    public void setData(int data) {this.data = data;}

    public Node getNextNode(){return next;}

    public void setNextNode(Node next){this.next = next;}
}

class Queue{
    Node front,rear;
    
    public Queue(){
        front = rear = null;
    }

    public boolean isEmpty(){
        return (rear == null || front == null);
    }

    public void EnQueue(int data){
        Node newNode = new Node(data);
        
        if(isEmpty()){
            front = rear = newNode;
            return;
        }

        rear.setNextNode(newNode);
        rear = newNode;
    }

    public int DeQueue(){
        if(isEmpty()){
            System.out.println("Queue is empty");
            return -1;
        }

        int data = front.getData();

        if(front.getNextNode() == null)
            front = rear = null;
        else front = front.getNextNode();

        return data;
    }

    public static void main(String []arg){
        Queue q = new Queue();
        q.EnQueue(1);
        q.EnQueue(2);
        q.EnQueue(3);
        q.EnQueue(4);
        q.EnQueue(5);
        q.EnQueue(6);

        System.out.println(q.DeQueue());
        System.out.println(q.DeQueue());
        System.out.println(q.DeQueue());
        System.out.println(q.DeQueue());
        System.out.println(q.DeQueue());
        System.out.println(q.DeQueue());
        System.out.println(q.DeQueue());
    }
}

File: Queue/Queue.c
#include<stdio.h>
#include<stdlib.h>

// int *Queue;
// int front,rear,n;

// int is_empty(){
//     return (front == -1 || rear == -1);
// }

// void enqueue(int data){
//     if(is_empty()){
//         front=rear=0;
//         Queue[rear] = data;
//         return;
//     }
//     if(rear == n-1){
//         printf("\nQueue extend limits\n");
//         return;
//     }
//     Queue[++rear] = data;
// }

// int dequeue(){
//     if(is_empty()){
//         printf("\nQueue is empty\n");
//         return -1;
//     }

//     if(front == rear){
//         int data = Queue[front];
//         front = rear = -1;
//         return data;
//     }

//     return Queue[front++];
// }

// int main(){
//     printf("\nEnter a limit of Queue: ");
//     scanf("%d",&n);
    
//     Queue = (int*)malloc(n*sizeof(int));
//     rear = front = -1;

//     enqueue(1);
//     enqueue(2);
//     enqueue(3);
//     enqueue(4);
//     enqueue(5);
//     enqueue(6);
//     enqueue(7);

//     printf("%d\n",dequeue());
//     printf("%d\n",dequeue());
//     printf("%d\n",dequeue());
//     printf("%d\n",dequeue());
//     printf("%d\n",dequeue());
//     printf("%d\n",dequeue());
//     printf("%d\n",dequeue());
//     printf("%d\n",dequeue());
// }

                            // linkedlist module of Queue

struct Queue_node{
    int data;
    struct Queue_node* next;
}*front,*rear;

void enqueue(int data);

int dequeue();

int is_empty();

int main(){
    front = rear = NULL;
    enqueue(1);
    enqueue(2);
    enqueue(3);
    enqueue(4);
    enqueue(5);
    enqueue(6);
    enqueue(7);

    printf("%d\n",dequeue());
    printf("%d\n",dequeue());
    printf("%d\n",dequeue());
    printf("%d\n",dequeue());
    printf("%d\n",dequeue());
    printf("%d\n",dequeue());
    printf("%d\n",dequeue());
    printf("%d\n",dequeue());
}

int is_empty(){
    return (rear == NULL || front == NULL);
}

void enqueue(int data){
    struct Queue_node *newNode;
    newNode = (struct Queue_node*)malloc(sizeof(struct Queue_node));
    newNode->data = data;
    
    if(is_empty()){
        front = rear = newNode;
        return;
    }

    rear->next = newNode;
    rear = newNode;
}

int dequeue(){
    if(is_empty()){
        printf("\nQueue is empty\n");
        return -1;
    }

    struct Queue_node *temp = front;

    if(front->next == NULL)
        front = rear = NULL;
    else front = front->next;

    int data = temp->data;
    free(temp);
    return data;
}